app/Controllers/Http/UsersController.ts

```diff
@@ -0,0 +1,54 @@
+import { HttpContextContract } from '@ioc:Adonis/Core/HttpContext';
+import User from 'App/Models/User';
+import CreateUser from 'App/Validators/CreateUserValidator';
+import LoginUser from 'App/Validators/LoginUserValidator';
+
+export default class UsersController {
+    public async register({ request, response }: HttpContextContract) {
+        const user = new User();
+        const payload = await request.validate(CreateUser);
+
+        user.name = payload.name;
+        user.email = payload.email;
+        user.password = payload.password;
+
+        await user.save();
+        response.status(201);
+
+        if(!user.$isPersisted) {
+            return {
+                status: 'fail',
+                message: 'Failed to register user'
+            }
+        }
+
+        return {
+            status: 'success',
+            data: user,
+            message: 'User registered successfully'
+        }
+    }
+
+    public async login({ auth, request }: HttpContextContract) {
+        const { email, password } = await request.validate(LoginUser);
```

Comment:
I like this approach of validation, that is nice!
Is the validate function baked on top of the API request object by adonis?

Reply:
Yes, it is baked on top of the API request object by adonis as it validates all incoming input.